core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -455,8 +459,7 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
         this.activeSuggestion.target_id, this.activeSuggestionId,
         AppConstants.ACTION_REJECT_SUGGESTION,
         reviewMessage || this.reviewMessage, null,
-        this.resolveSuggestionAndUpdateModal.bind(this),
-        (error) => {
+        onSuccess, (error) => {
```

Comment:
Why not just declare the `onSuccess` function inline here instead of passing it in as an argument?

Reply:
@sagangwee Because `acceptAndReviewNext` and `rejectAndReviewNext` were to be implemented with different behaviors when the rejection of a suggestion was successful. For `rejectAndReviewNext`, it displays a success message to the reviewer letting the individual know that the rejection of said suggestion was a success. For `acceptAndReviewNext`, it would have already displayed a warning to the reviewer that the suggestion acceptance failed backend validation checks, before rejecting the suggestion. I thought it would have been odd for a success message stating "Suggestion rejected." to show up out of the blue without the program giving a heads-up to the reviewer to reject the suggestion. I had several solutions at the time, but all had potential issues with being properly read by the reviewer:

1. Change warning message to `"Invalid suggestion: {error message}. Rejecting..."` Depending on the length of the `error message`, the reviewer might not have enough time to read the "Rejecting..." part at the end of the message before the alert goes away.
2. Add an info message that displays after the warning. The info message states "Rejecting...". While this makes the "Rejecting..." clear to the reviewer, it also adds to the number of alerts on the reviewer's screen at a time. I believe warnings and success/info messages are also displayed on different parts of the screen (warnings at the top & messages at the bottom right corner), so this causes conflict for the reviewer's eyes.

I ended up choosing to reject the invalid suggestion silently, so as to avoid such readability issues.

Reply:
Ah, I didn't realize this function was being called more than once. I don't think we should actually reject during an accept error anyway. Left a comment above.